examples/routing_service/routing_service_file_adapter/cpp/FileStreamReader.cxx

```diff
@@ -0,0 +1,146 @@
+        if (input_file_stream_.is_open()) {
+            std::getline(input_file_stream_, buffer_);
+
+            /** 
+             * Here we notify routing service, that there is data available 
+             * on the StreamReader, triggering a call to take().
+             */
+            if (!input_file_stream_.eof()) {
+                reader_listener_->on_data_available(this);
+            } else {
+                stop_thread_ = true;
+            }
+        }
+
+        std::this_thread::sleep_for(std::chrono::seconds(sampling_period_));
+    }
+    std::cout << "Reached end of stream for file " << input_file_name_ << std::endl;
+    file_connection_->dispose_discovery_stream(stream_info_);
+}
+
+FileStreamReader::FileStreamReader(
+        FileConnection *connection, 
+        const StreamInfo &info,
+        const PropertySet &property,
+        StreamReaderListener *listener)
+        : filereader_thread_(),
+        sampling_period_(1), 
+        stop_thread_(false), 
+        stream_info_(info.stream_name(), info.type_info().type_name())
+{
+    file_connection_ = connection;
+    reader_listener_ = listener;
+    adapter_type_ = static_cast<DynamicType *>(
+            info.type_info().type_representation());
+
+    // Parse the properties provided in the xml configuration file
+    for (const auto &it : property) {
+        if (it.first == INPUT_FILE_PROPERTY_NAME) {
+            input_file_name_ = it.second;
+            input_file_stream_.open(it.second);
+        } else if (it.first == SAMPLE_PERIOD_PROPERTY_NAME) {
+            sampling_period_ = std::stoi(it.second);
+        }
+    }
+
+    if (!input_file_stream_.is_open()) {
+        throw dds::core::IllegalOperationError(
+                "Input file not provided or unable to open");
+    } else {
+        std::cout << "Input file name: " << input_file_name_ << std::endl;
+    }
+
+    filereader_thread_ = std::thread(&FileStreamReader::file_reading_thread, this);
+}
+
+void FileStreamReader::take(
+        std::vector<dds::core::xtypes::DynamicData *> &samples,
+        std::vector<dds::sub::SampleInfo *> &infos)
+{
+    std::istringstream s(buffer_);
+    std::string color;
+    std::string x;
+    std::string y;
+    std::string shapesize;
+
+    // Reading input from the csv file
+    std::getline(s, color, ',');
+    std::getline(s, x, ',');
+    std::getline(s, y, ',');
+    std::getline(s, shapesize, ',');
+
+    samples.resize(1);
+    infos.resize(1);
+
+    DynamicData *sample = new DynamicData(*adapter_type_);
+
+    /** 
+     * This is the hardcoded type information about ShapeType.
+     * You are advised to change this as per your type definition
+     */
+    sample->value("color", color);
+    sample->value("x", std::stoi(x));
+    sample->value("y", std::stoi(y));
+    sample->value("shapesize", std::stoi(shapesize));
+
+    dds::sub::SampleInfo *info = new dds::sub::SampleInfo();
+    (*info)->native().valid_data = DDS_BOOLEAN_TRUE;
```

Comment:
Is it necessary to create a SampleInfo object just to set the valid_data flag to true? Shouldn't it work if you leave the info sequence empty or set the elements to null?

Reply:
IDK if the valid_data flag is set to true by default. Even if it is so, it would be helpful to demonstrate the use, in case someone wants to use the flag to indicate invalid data.

Reply:
Actually returning SampleInfo is not needed, so we can just leave the info_seq empty.